src/orders/components/OrderCancelDialog/OrderCancelDialog.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { OrderErrorCode, OrderErrorFragment } from "@dashboard/graphql";
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { FormattedMessageProps } from "react-intl";
+
+import { OrderCancelDialog } from "./OrderCancelDialog";
+
+jest.mock("react-intl", () => ({
+  useIntl: jest.fn(() => ({
+    formatMessage: jest.fn(x => x.defaultMessage),
+  })),
+  defineMessages: jest.fn(x => x),
+  FormattedMessage: ({ defaultMessage, values }: FormattedMessageProps) => {
+    let formattedMessage = defaultMessage;
+
+    // Replace placeholders with corresponding values if values are provided
+    if (values) {
+      Object.keys(values).forEach(key => {
+        const valueRegex = new RegExp(`#{${key}}`, "g");
+        formattedMessage = formattedMessage.replace(
+          valueRegex,
+          values[key] as string,
+        );
+      });
+    }
+
+    return <>{formattedMessage}</>;
+  },
```

Comment:
question: why do you do this, it wouldn't be enough to mock like this
```
  FormattedMessage: ({ defaultMessage }: { defaultMessage: string }) => (
    <>{defaultMessage}</>
  ),
  ```

Reply:
It's not enough unfortunately, the first test won't pass, because dialog title will be rendered as `Order number {orderNumber}` and we look for `456` in `expect` statement

Reply:
IMO checking text `Order number {orderNumber}` is enough.